src/kudu/consensus/raft_consensus.cc

```diff
@@ -3586,6 +3586,7 @@ void RaftConsensus::HandleProxyRequest(const ConsensusRequestPB* request,
 
     vector<ReplicateRefPtr> messages;
     do {
+      messages.clear();
```

Comment:
can we break down this massive function into smaller pieces?

Reply:
Yes that needs to happen but it will need to happen anyway for nonblocking support so best to do it once at that time.